lib/iaas/index.js

```diff
@@ -2,15 +2,20 @@
 
 const config = require('../config');
 const CloudProviderClient = require('./CloudProviderClient');
+const AzureClient = require('./AzureClient');
 const BackupStore = require('./BackupStore');
 const BackupStoreForServiceInstance = require('./BackupStoreForServiceInstance');
 const BackupStoreForOob = require('./BackupStoreForOob');
+const BaseCloudClient = require('./BaseCloudClient');
 
-const cloudProvider = new CloudProviderClient(config.backup.provider);
+const cloudProvider = config.backup.provider.name === 'azure' ?
```

Comment:
Better to move this conditional creation into `BaseCloudClient` new factory method which provides right implementation based on input provider name.

Reply:
done